### ISSUE TITLE
Add info about Git Credential manager and how to disable it to FAQ

### ISSUE THREAD

**Issue by SteveALee** [git-for-windows/git#745]
Having installed the latest GfW after a PC reset I find my SSH agent is being broken by Git Credential manager. After conciderable searching and faffing about I discovered how to disable int the system config.

Please add to FAQ and also make such a breaking change obvious on youe home page and release notes.

thanks

**Comment by SteveALee** [comment]
In fact I have not been able to get ssh agent to work again. Please explain what I need to do to restore previous behaviour. Just renaming the git configfile is not enough

**Comment by shiftkey** [comment]
@SteveALee it sounds like you don't have a credential helper defined any more after renaming the system config file. To confirm this, run:

> git config credential.helper

I'm not sure if the previous `wincred` provider is available, but you can try setting this command to see if you continue to be prompted by ssh-agent:

> git config --global credential.helper wincred

**Comment by SteveALee** [comment]
Ah, so you need one set? I'll try the old one and if not will have to roll back to a previous version.

Ssh agent never prompts, that's what's so good about it. It just uses your private keys.

Thanks.

**Comment by shiftkey** [comment]
@SteveALee I'm not clear on what ssh-agent prompts you're referring to here, so perhaps there's some other config here that's been removed - could you give an example of what you're seeing?

**Comment by SteveALee** [comment]
Until now I have never had any prompts. I configured SSH Agent to run and on the first GfW bash run after reboot I see this text

```
Initializing new SSH agent...
succeeded
Identity added: /c/Users/steve/.ssh/id_rsa (/c/Users/steve/.ssh/id_rsa)
```

This has been so for a long time. THen when I ```push``` it just works without any prompts. SSHAgent is providing my SSH keys to git.

However after the latest install of GfW (I assume this the culprit) I now get interfering UI for GCM whenever I push.

It is of course possible I did not restore somehting but most unlikey. Due to ongoing problems with my Surface Pro 4 I have reset and restored it several times. Previously there were no issues.

**Comment by SteveALee** [comment]
@shiftkey 

> you can try setting this command to see if you continue to be prompted by ssh-agent:
>
>    git config --global credential.helper wincred

I still get the GCM UI and then when I exit

```$ git push
Logon failed, use ctrl+c to cancel basic credential prompt.
Username for 'https://github.com/':
```

and then Ctrl-C gives me

```
steve@OD-Surface MINGW64 /c/projects/brian (master)
$ bash: line 0: read: read error: 0: Input/output error
```

**Comment by SteveALee** [comment]
@shiftkey 



>>    you can try setting this command to see if you continue to be prompted by ssh-agent:
>>
>>    git config --global credential.helper wincred
>>
>
>I still get the GCM UI and then when I exit

however if I edit

C:\Program Files\Git\mingw64\etc\gitconfig

i lose the UI but get a prompt in bash

This sort of indicates "system" is taking precidence over "global". Is that expected?

**Comment by SteveALee** [comment]
@shiftkey 

finally FYI here is my .bashrc for running ssh-agent.exe

```
SSH_ENV=$HOME/.ssh/environment
   
# start the ssh-agent
function start_agent {
    echo "Initializing new SSH agent..."
    # spawn ssh-agent
    /usr/bin/ssh-agent | sed 's/^echo/#echo/' > "${SSH_ENV}"
    echo succeeded
    chmod 600 "${SSH_ENV}"
    . "${SSH_ENV}" > /dev/null
    /usr/bin/ssh-add
}
   
if [ -f "${SSH_ENV}" ]; then
     . "${SSH_ENV}" > /dev/null
     ps -ef | grep ${SSH_AGENT_PID} | grep ssh-agent$ > /dev/null || {
        start_agent;
    }
else
    start_agent;
fi
```

**Comment by SteveALee** [comment]
So it appears ssh-agent was still working but the other prompts simply got in the way. I've disabled WCM by re-installing GfW and unchecking the option.

However I'm still getting a prompt for user in the console and do not seem to have any reference to wincred. ```git config credential.helper``` returns nothing. How do I get rid of this unwanted prompt?

Thanks for all your help

**Comment by shiftkey** [comment]
@SteveALee it's hard to say from here based on those new details, but this suggests the remote for your repository is setup to use HTTP instead of a SSH URL:

```
$ git push
Logon failed, use ctrl+c to cancel basic credential prompt.
Username for 'https://github.com/':
```

That would explain much of what you're seeing here.

Can you run these commands from the failing repository and attach the output:

> git config -l --system
> git config -l --global
> git config -l 
> git remote -vv

**Comment by dscho** [comment]
> it's hard to say from here based on those new details, but this suggests the remote for your repository is setup to use HTTP instead of a SSH URL

I agree that the info is quite a bit scarce and it did not help that the reporter chose to delete the issue reporting template instead of filling it out. In particular a [Minimal, Complete & *Verifiable* Example](http://stackoverflow.com/help/mcve) is sorely missing.

Further, it is indeed the case that `ssh` has no way to call the Git Credential Manager. So it seems that the reported problem is not due to a bug.

**Comment by SteveALee** [comment]
Oh FFS! You are correct. I'm not sure how that changed during my many PC restores. I must have checked out rather than use a backup. Such an easy thing to overlook given the messages don't completely spell it out for dummies like me :) To be honest though I saw no indication that GCM is for HTTP only :(

I just changed the remote and all is well.
I previously rolled back as far as 2.6.0 and still got prompted. That was a clue it was not G4W. But I had convinced myself the config was unchanged. Didn't think of the remotes

Thank you!

Apologies for not using the template - it was **very** detailed and I'm under huge pressure to get this working.

Thanks again for your time
